compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1674,6 +1674,12 @@ public BType readType(int cpI) throws IOException {
                     if (Symbols.isFlagOn(flags, Flags.ANONYMOUS)) {
                         objectSymbol.flags |= Flags.ANONYMOUS;
                     }
+                    if (Symbols.isFlagOn(flags, Flags.DISTINCT)) {
```

Comment:
Wouldn't this approach still leave room for error? Can't we read/write the symbol flags and set that directly (if `flags` type flags are different)?

Reply:
Do you mean at the type writer in a place like [this](https://github.com/ballerina-platform/ballerina-lang/blob/d08328e9b726c32553ce6bf9a5f2c70412a44e86/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java#L449) we should write the symbol flags value to the BIR. In the BIRSymbolEnter we should read the flag values and set in the symbol, rather than reading flags one by one.

Reply:
Yeah?

Reply:
We need to check if this can cause any issues (not that I can think of any) since it will obviously mean more flags.

Reply:
I'll try this approach and give an update.

Reply:
The approach you have suggested is working.